proof/crefine/AARCH64/Fastpath_Defs.thy

```diff
@@ -41,37 +41,22 @@ definition
     newVTable \<leftarrow> liftE $ getThreadVSpaceRoot dest >>= getCTE;
     unlessE (isValidVTableRoot $ cteCap newVTable) $ throwError ();
 
-\<comment> \<open> (* FIXME AARCH64 TODO *) AArch64+hyp specific tests
-    /* Need to test that the ASID is still valid */
-    asid_t asid = cap_vspace_cap_get_capVSMappedASID(newVTable);
-    asid_map_t asid_map = findMapForASID(asid);
-    if (unlikely(asid_map_get_type(asid_map) != asid_map_asid_map_vspace ||
-                 VSPACE_PTR(asid_map_asid_map_vspace_get_vspace_root(asid_map)) != cap_pd)) {
-        slowpath(SysCall);
-    }
-    /* Ensure the vmid is valid. */
-    if (unlikely(!asid_map_asid_map_vspace_get_stored_vmid_valid(asid_map))) {
-        slowpath(SysCall);
-    }
-    /* vmids are the tags used instead of hw_asids in hyp mode */
-    stored_hw_asid.words[0] = asid_map_asid_map_vspace_get_stored_hw_vmid(asid_map);
-\<close>
-
-\<^cancel>\<open> this check appears to only happen on AARCH32
-    pd \<leftarrow> returnOk $ \<^cancel>\<open>capPTBasePtr\<close> XXX $ capCap $ cteCap newVTable;
-\<close>
+    vspace_root \<leftarrow> returnOk $ capPTBasePtr $ capCap $ cteCap newVTable; \<comment> \<open>cap_pd in C\<close>
+    asid \<leftarrow> returnOk $ (fst $ the $ capPTMappedAddress $ capCap $ cteCap newVTable);
```

Comment:
```suggestion
    asid \<leftarrow> returnOk $ fst $ the $ capPTMappedAddress $ capCap $ cteCap newVTable;
```
(or leaving out the first `$` instead should also work)

Reply:
done